### PR TITLE
Hearings/Reader link

### PR DESCRIPTION
**Pull request by sharonwarner**
This PR 
- Adds a new category for documents called case_summary
- Pre-sets the doc category to true if the document has a case_summary type
- Allows the user to filter on case_summary
- Doesn't allow the user to reset the case_summary category on docs

screen shot 2017-08-01 at 4 53 04 pm

screen shot 2017-08-01 at 4 52 54 pm

Connects #2137

### CODE REVIEW

File: client/app/reader/constants.js [hunk 1]
```diff
@@ -94,17 +94,23 @@ export const documentCategories = {
...
-    svg: <CategoryIcon color="#4A90E2" />
+    svg: <CategoryIcon color="#5A94EC" />
   },
   medical: {
     renderOrder: 1,
     humanName: 'Medical',
     svg: <CategoryIcon color="#FF6868" />
   },
-  other: {
+  case_summary: {
```
Comment (mdbenjam) [inline client/app/reader/constants.js]: I think you should put this as the first or last category. Not in the middle. It looks weird to have one non-clickable checkbox in the middle of the list.

File: db/migrate/20170727203829_add_category_to_documents.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class AddCategoryToDocuments < ActiveRecord::Migration
+  def change
+    add_column :documents, :category_case_summary, :boolean
```
Comment (mdbenjam) [inline db/migrate/20170727203829_add_category_to_documents.rb]: Actually, I've been thinking about this. Not sure we actually need to store this information. This is known already based on the document type. I think we should do it dynamically. Two main reasons.

1. What happens when someone changes a document type. Starts as Form 9--we mark it as case summary, then it's changed to Form X. It's still marked.
1. What happens for documents already in our DB? If we go this route, we should make this a more complex migration and check the appropriate boxes for all those cases.

File: app/controllers/document_controller.rb [hunk 3]
```diff
@@ -46,7 +46,7 @@ def mark_as_read
   DOCUMENT_AUTHORIZED_ROLES = ["Reader"].freeze
 
   def update_params
-    params.permit(:category_procedural, :category_medical, :category_other)
+    params.permit(:category_procedural, :category_medical, :category_case_summary, :category_other)
```
Comment (mdbenjam) [inline app/controllers/document_controller.rb]: Do we need this here? We don't actually want to accept this parameter.

File: client/app/components/Checkbox.jsx [hunk 4]
```diff
@@ -35,6 +36,7 @@ export default class Checkbox extends React.Component {
           type="checkbox"
           id={name}
           checked={value}
+          disabled={disabled}
```
Comment (mdbenjam) [inline client/app/components/Checkbox.jsx]: Something looks weird here, the checkbox shifts. Not worth holding up--but I can't figure out why.
weirdcheckbox

File: client/app/reader/constants.js [hunk 5]
```diff
@@ -102,7 +102,13 @@ export const documentCategories = {
   other: {
     renderOrder: 2,
     humanName: 'Other Evidence',
-    svg: <CategoryIcon color="#5BD998" />
+    svg: <CategoryIcon color="#3AD2CF" />
+  },
+  case_summary: {
+    renderOrder: 3,
+    humanName: 'Case Summary',
+    svg: <CategoryIcon color="#FDC231" />,
```
Comment (mdbenjam) [inline client/app/reader/constants.js]: I'm loving the new colors!

**Comment by NickHeiner** [comment]
image

Wow, I'm impressed that this only took 62 added lines. :smile:

File: client/app/reader/DocCategoryPicker.jsx [hunk 6]
```diff
@@ -17,19 +17,21 @@ const CategorySelector = (props) => {
   };
 
   return <Checkbox name={categoryName} onChange={handleChange}
-      label={label} value={toggleState} />;
+      label={label} value={toggleState} disabled={category.readOnly && allowReadOnly} />;
```
Comment (NickHeiner) [inline client/app/reader/DocCategoryPicker.jsx]: Why is `allowReadOnly` necessary?

**Comment by NickHeiner** [comment]
I had one small comment, and it's not blocking.

File: client/app/reader/DocCategoryPicker.jsx (hunk 6 above)
Comment (sharonwarner) [inline client/app/reader/DocCategoryPicker.jsx]: We use the same component for the checkboxes that allows users to filter by category on the homepage and the checkboxes that allow users to change the categories on the document view page. We need the checkbox to be readOnly for the second case, but not readOnly for the first case.

**Pull request merged**